WinCCOA_QualityChecks/scripts/libs/classes/QualityGates/QgStaticCheck/Panels/PanelFile/PanelFile.ctl

```diff
@@ -50,98 +64,111 @@ class PanelFile
     
     int rc = system(cmd);
     if ( rc )
+    {
+      logger.warning("Can not convert panel into xml format: " + cmd);
       return -2;
+    }
     
     return 0;
   }
   
+  //------------------------------------------------------------------------------
   public bool isBackUp()
   {
     return (strpos(getExt(_relPath), "bak") == 0);
   }
   
+  //------------------------------------------------------------------------------
   public string getFullPath()
   {
     return _sourceDir + PANELS_REL_PATH + _relPath;
   }
   
+  //------------------------------------------------------------------------------
   public string getRelPath()
   {
     return _relPath;
   }
   
+  //------------------------------------------------------------------------------
   public int read()
   {
-    if ( !exists() )
-    {
-      DebugFTN("PanelFile", __FUNCTION__, "panel does not exists", getFullPath());
-      return -1;
-    }
+    File panelFile = File(getFullPath());
     
-    fileToString(getFullPath(), strContent);
+    panelFile.read(this.strContent);
     
-    if ( strContent != "" )
+    if ( this.strContent != "" )
     {      
-      _isXml = (strpos(strContent, "<?xml version=") == 0);
-      _isCrypted = (strpos(strContent, "PVSS_CRYPTED_PANEL") == 0);
+      _isXml = this.strContent.startsWith("<?xml version=");
```

Comment:
Shouldn't we change this to more error-proof XML-parsing like RegExp:
<\?\s*xml\s+version\s*=